discord_bots/cogs/category.py

```diff
@@ -19,23 +20,63 @@ def __init__(self, bot: Bot):
 
     group = app_commands.Group(name="category", description="Category commands")
 
-    @group.command(name="create", description="Create a new category")
+    @group.command(name="configure", description="Create/Edit a category")
     @app_commands.check(is_admin_app_command)
     @app_commands.check(is_command_channel)
-    @app_commands.describe(name="Name of new category")
-    async def createcategory(self, interaction: Interaction, name: str):
+    @app_commands.describe(category_name="New or existing category")
+    async def configure(self, interaction: Interaction, category_name: str):
+        assert interaction.guild
+
+        new_category: bool = False
         session: SQLAlchemySession
         with Session() as session:
-            session.add(Category(name=name, is_rated=True, sigma_decay_amount=default_sigma_decay_amount()))
-            session.commit()
+            category: Category | None = (
+                session.query(Category)
+                .filter(Category.name.ilike(category_name))
+                .first()
+            )
+            if not category:
+                new_category = True
+                category = Category(
+                    name=category_name,
+                    is_rated=True,
+                    sigma_decay_amount=default_sigma_decay_amount(),
+                )
+
+            configure_view = CategoryConfigureView(interaction, category)
+            configure_view.embed = Embed(
+                description=f"**{category.name}** Category Configure\n-----",
+                colour=Colour.blue(),
+            )
             await interaction.response.send_message(
-                embed=Embed(
-                    description=f"Category **{name}** added",
-                    colour=Colour.green(),
-                ),
+                embed=configure_view.embed,
+                view=configure_view,
                 ephemeral=True,
             )
 
+            await configure_view.wait()
+            if configure_view.value:
+                if new_category:
+                    session.add(category)
+                session.commit()
+                await interaction.delete_original_response()
```

Comment:
I thought you weren't able to delete ephemerally sent messages, but I think I'm wrong on that one.

Instead of deleting, why not just leave the view so that the user can keep on editing for as long as they like? That way you don't need to consider this operation as "one and done" and can instead update the category table every time they press submit.